tests/Basics/InspectionTest.cpp

```diff
@@ -298,6 +300,42 @@ struct Specialization {
   int i;
   std::string s;
 };
+
+enum class AnEnumClass { Option1, Option2, Option3 };
+
+auto to_string(AnEnumClass e) -> std::string_view {
+  switch (e) {
+    case AnEnumClass::Option1:
+      return "Option1";
+    case AnEnumClass::Option2:
+      return "Option2";
+    case AnEnumClass::Option3:
+      return "Option3";
+  }
+  return "invalid.";
+}
+
+template<typename Enum>
+struct EnumStorage {
+  using MemoryType = Enum;
+
+  std::underlying_type_t<Enum> code;
+  std::string message;
+
+  explicit EnumStorage(Enum e)
+      : code(static_cast<std::underlying_type_t<Enum>>(e)),
+        message(to_string(e)){};
+  explicit EnumStorage() {}
+
+  operator Enum() const { return Enum(code); }
+};
+
+template<class Inspector, class Enum>
+auto inspect(Inspector& f, EnumStorage<Enum>& e) {
+  return f.object(e).fields(f.field("code", e.code),
+                            f.field("message", e.message).fallback(""));
```

Comment:
This works but actually loads the message just so we discard it afterwards. Alternatively you could do it like this:
```
if constexpr (Inspector::isLoading) {
  return f.object(e).fields(f.field("code", e.code),
                            f.ignoreField("message");
} else {
  return f.object(e).fields(f.field("code", e.code),
                            f.field("message", e.message));
}
```